src/views/GroupView.vue

```diff
@@ -1,73 +1,669 @@
+              attribution:
+                '<a href="//www.linz.govt.nz/data/linz-data/linz-basemaps/data-attribution">LINZ CC BY 4.0 © Imagery Basemap contributors</a>',
+              url:
+                `https://basemaps.linz.govt.nz/v1/tiles/aerial/3857/{z}/{x}/{y}.webp?api=${linzBasemapApiKey}`,
+            },
+            {
+              name: "OpenStreetMap Basemap",
+              visible: false,
+              url: "https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png",
+              attribution:
+                '&copy; <a href="http://osm.org/copyright">OpenStreetMap</a> contributors',
+            },
+          ],
+          icon: Marker,
+        },
+        // Stations state
+        backDateRecordings: false,
+        applyStationsFromDate: null,
+        stationsLoading: false,
+        addingStations: false,
+        items: [],
+        pendingStations: [],
+        draggingCsvOver: false,
+        enableEditingStations: true,
+      },
+      isLoading: false, // Loading all data on page load
+      group: null,
+      currentTabIndex: 0, // Which tab is the default one
+      showUserRemoveSelfModal: false,
+      isRemovingUser: false,
     };
   },
-  computed: mapState({
-    group: (state) => state.Groups.currentGroup,
-    fetched: (state) => state.Groups.fetched,
-    currentUser: (state) => state.User.userData,
-  }),
-  watch: {
-    $route() {
-      this.fetchGroup();
+  computed: {
+    ...mapState({
+      currentUser: (state) => (state as any).User.userData,
+    }),
+    groupName() {
+      return this.$route.params.groupname;
+    },
+    isGroupAdmin() {
+      if (
+        (this.currentUser && this.currentUser.isSuperUser) ||
+        (this.group && this.group.Users)
+      ) {
+        return (
+          this.currentUser.isSuperUser ||
+          this.groupUsers.some(
+            (user) =>
+              user.username === this.currentUser.username && user.isAdmin
+          )
+        );
+      }
+      return false;
+    },
+    groupUsers() {
+      return (this.group && this.group.GroupUsers) || [];
+    },
+    groupDevices() {
+      return (this.group && this.group.Devices) || [];
+    },
+    groupHasDevices() {
+      return this.groupDevices.length !== 0;
+    },
+    groupHasUsers() {
+      return this.groupUsers.length !== 0;
+    },
+    groupHasStations() {
+      return this.stations.items.length !== 0;
+    },
+    stationsForMap() {
+      // Stations lat/lng as leaflet lat/lng objects
+      return this.stations.items.map(({ name, latitude, longitude }) => ({
+        name,
+        location: latLng(latitude, longitude),
+      }));
+    },
+    mapBounds() {
+      // Calculate the initial map bounds and zoom level from the set of lat/lng points
+      return latLngBounds(this.stationsForMap.map(({ location }) => location));
+    },
+    pendingStationsDiff() {
+      // Show pending stations, and mark any existing stations that don't have a match in pending as
```

Comment:
Hmm we have now duplicated this logic in the api and web.   So it would be easy to get out of sync.